00-basics/002-javascript/Kristina/Learning002.js

```diff
@@ -0,0 +1,143 @@
+let RobotDinosaurFromSpace = {};
+RobotDinosaurFromSpace.roar = function(line) {
+    console.log(`The creature roars '${line}'`);
+};
+
+RobotDinosaurFromSpace.roar("I shall eat all of you puny humans ");
+function roar (line) {
+    console.log(`The ${this.type} Robot Dinosaur From Space Roars '${line}'`);
+}
+let EvilDinosaur= {type: "mean", roar};
+let SemiDecentDinosaur= {type: "bored", roar};
+
+EvilDinosaur.roar("I shall eat all you puny humans," +
+    "suffer in my jaws of death!");
+
+SemiDecentDinosaur.roar("....Let's at least knock them out first I don't like when my food screams");
+roar.call(EvilDinosaur, "Burp! " + "*barely audible screetching*");
+roar.call(SemiDecentDinosaur, "....I hate you and I have a migraine");
+let protoDino= {
+    roar(line) {
```

Comment:
wait, is this even correct syntax? I would imagine JS complaining because there is no assignment. This would work if it was a class, but for an object it would be
```
{
roar: function(line) {
}
}
```